pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -669,17 +240,23 @@ func GenerateDeployment(ctx resource.Context, cfg echo.Config, settings *resourc
 		return "", err
 	}
 
-	deploy := deploymentTemplate
+	templateName := "echo_deployment.template"
 	if cfg.DeployAsVM {
-		deploy = vmDeploymentTemplate
+		templateName = "vm_deployment.template"
+	} else if os.Getenv("DEPLOYMENT_TEMPLATE") != "" {
```

Comment:
Let's make this a flag instead of an environment variable. For completeness, let's add the following:

- `istio.test.echo.kube.template.service`: name of the file to be used for the echo service. If not specified, use the current default service file.
- `istio.test.echo.kube.template.deployment`: name of the file to use by default for non-vms. If not specified, use the current default file name.
- `istio.test.echo.kube.template.deployment.vm`: name of the file to use by default for VMs. If not specified, use the current default file name.

For each of these flags, allow absolute paths to be specified. If the string is not an absolute file path, assume it's relative to the `templates` directory. When we startup, we only have to read the files from the flags, not all files in the directory.